st2api/st2api/listener.py

```diff
@@ -116,3 +122,8 @@ def get_listener():
             _listener = Listener(conn)
             eventlet.spawn_n(listen, _listener)
     return _listener
+
+
+def get_listener_if_set():
```

Comment:
I've added this function so we can use it in st2api/cmd/api.py and do something like this on CTRL+C (SIGINT).

```python
listener = get_listener_if_set()

if listener:
    listener.shutdown()
```